app/api/competitions/route.js

```diff
@@ -0,0 +1,32 @@
+import dbConnect from '@/utils/dbConnects';
+import Competition from '../models/Competition';
+import { getUserFromRequest } from '@/utils/auth';
+
+export async function GET() {
+  await dbConnect();
+  const competitions = await Competition.find().sort({ createdAt: -1 });
+  return Response.json(competitions);
+}
+
+export async function POST(req) {
+  try {
+    await dbConnect();
+
+    const user = await getUserFromRequest(req);
+    if (!user) {
+      return Response.json({ error: 'Unauthorized' }, { status: 401 });
+    }
+
+    const competitionData = await req.json();
```

Comment:
nit; consider adding validation to validate the input from the user is valid. This can be done later though